packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -509,28 +512,32 @@ Future<void> _buildGradleProjectV2(
     } else {
       appSize = ' (${getSizeAsMB(bundleFile.lengthSync())})';
     }
-    printStatus('Built ${fs.path.relative(bundleFile.path)}$appSize.');
+    printStatus('Built ${fs.path.relative(bundleFile.path)}$appSize.',
+        color: TerminalColor.green);
   }
 }
 
-File _findApkFile(GradleProject project, BuildInfo buildInfo) {
-  final String apkFileName = project.apkFileFor(buildInfo);
-  if (apkFileName == null)
-    return null;
-  File apkFile = fs.file(fs.path.join(project.apkDirectory.path, apkFileName));
-  if (apkFile.existsSync())
-    return apkFile;
-  final String modeName = camelCase(buildInfo.modeName);
-  apkFile = fs.file(fs.path.join(project.apkDirectory.path, modeName, apkFileName));
-  if (apkFile.existsSync())
-    return apkFile;
-  if (buildInfo.flavor != null) {
-    // Android Studio Gradle plugin v3 adds flavor to path.
-    apkFile = fs.file(fs.path.join(project.apkDirectory.path, buildInfo.flavor, modeName, apkFileName));
+Iterable<File> _findApkFiles(GradleProject project, BuildInfo buildInfo) {
+  final Iterable<String> apkFileNames = project.apkFilesFor(buildInfo);
+  if (apkFileNames.isEmpty)
+    return <File>[];
+
+  return apkFileNames.map<File>((String apkFileName) {
+    File apkFile = fs.file(fs.path.join(project.apkDirectory.path, apkFileName));
     if (apkFile.existsSync())
       return apkFile;
-  }
-  return null;
+    final String modeName = camelCase(buildInfo.modeName);
+    apkFile = fs.file(fs.path.join(project.apkDirectory.path, modeName, apkFileName));
+    if (apkFile.existsSync())
+      return apkFile;
+    if (buildInfo.flavor != null) {
+      // Android Studio Gradle plugin v3 adds flavor to path.
+      apkFile = fs.file(fs.path.join(project.apkDirectory.path, buildInfo.flavor, modeName, apkFileName));
```

Comment:
With this many subdirectories, this way is actually shorter, but the other way allows chaining... I think the chaining is easier to read, but your call.

```dart
project.apkDirectory
    .childDirectory(buildInfo.flavor)
    .childDirectory(modeName)
    .childFile(apkFileName)
```

Reply:
+1